### DISCUSSION TITLE
Benchmark to compare demosaic methods

### DISCUSSION THREAD

**Discussion by hgrecco** [colour-science/colour-demosaicing#59] [General]
Do you have any suggestions about which datasets would be the best to compare the accuracy of demosaic methods?

**Comment by thomasmansencal** [comment]
Hi @hgrecco,

I would need to spend some time checking my library to find out standard datasets used to compare them but, typically, a dataset having both natural, e.g., vegetation, rocks, and man made objects, e.g., windows, fences, stripes, would work. Anything that is susceptible to cause aliasing and moire basically.

To compare images, one usually computes PSNR, we have an implementation in `colour`: https://colour.readthedocs.io/en/develop/generated/colour.utilities.metric_psnr.html#colour.utilities.metric_psnr

Here are two that came up on the first paper I looked at:

- https://r0k.us/graphics/kodak/
- https://web.comp.polyu.edu.hk/cslzhang/CDM_Dataset.htm

**Comment by hgrecco** [comment]
Thanks a lot for the recommendation. Is there a dataset with a ground truth image (ie. "real" RGB image, either simulated or acquired with filters) so I can generate from those a mosaic image; and in turn try the different algorithms?

**Reply by thomasmansencal** [reply]
The two I linked in my previous answer should work no?
